### ISSUE TITLE
<a> buttons have different cursor behaviour to <button> buttons.

### ISSUE THREAD

**Issue by bardiharborow** [twbs/bootstrap#21479]
232e86d removed `.btn { cursor: pointer; }`, but kept [`[role="button"] { cursor: pointer; }`](https://github.com/twbs/bootstrap/blob/daff021578d3afb9e8da4353038a6a5ad62343e1/scss/_reboot.scss#L237), which causes `<a class="btn" role="button">` to behave differently to `<button class="btn">`.

/cc @mdo

**Comment by mdo** [comment]
Hmm, yeah that's a tough one. Will noodle on it.

**Comment by KeitIG** [comment]
Btw, should `a.btn` still have a `cursor: pointer`? If I get [Buttons shouldn’t have a hand cursor](https://medium.com/simple-human/buttons-shouldnt-have-a-hand-cursor-b11e99ca374b#.nty7rdk6j) correctly, it's more about the shape than the action.

**Comment by patrickhlauke** [comment]
> Btw, should a.btn still have a cursor: pointer?

if it behaves like a button (triggers a behavior, rather than acting as a link to another page/part of the current page), then in theory yes. But as some authors will apply `.btn` for things like calls to action that are in fact links to another page, it's a tough one to untangle one way of another...

**Comment by patrickhlauke** [comment]
But in general, assuming authors give the correct `role="button"` to links that act as buttons, then you'd probably want something like

```
[role="button"] { cursor: default; }
```

and possibly even things like `role="tab"`

**Comment by vanduynslagerp** [comment]
This all the cases referenced [in Bootstrap docs](https://v4-alpha.getbootstrap.com/components/buttons/#button-tags)
```html
<a class="btn btn-primary" href="#" role="button">Link</a>
<button class="btn btn-primary" type="submit">Button</button>
<input class="btn btn-primary" type="button" value="Input">
<input class="btn btn-primary" type="submit" value="Submit">
<input class="btn btn-primary" type="reset" value="Reset">
```

I think the appropriate rule is:
```css
button,
[type="button"],
[type="reset"],
[type="submit"],
[role="button"] {
cursor: default;
}
```

**Comment by mdo** [comment]
Fixed with #21812.
